internal/background/availability_queue.go

```diff
@@ -31,12 +40,36 @@ func sendAvailabilityRequestMessages(ctx context.Context, batchSize int, tickDur
 	logger := ctxval.Logger(ctx)
 	ticker := time.NewTicker(tickDuration)
 	messageBuffer := make([]*kafka.GenericMessage, 0, batchSize)
+	sc, err := clients.GetSourcesClient(ctx)
+	if err != nil {
+		logger.Warn().Msg("Could not get sources client")
+		return
+	}
 
 	for {
 		select {
-		case msg := <-kafkaAvailabilityRequest:
-			messageBuffer = append(messageBuffer, msg)
+		case info := <-kafkaAvailabilityRequest:
+			ctx, err = ctxval.WithIdentityFrom64(ctx, info.identity)
+			if err != nil {
+				logger.Warn().Msg("Could not add identity header to ctx")
+				return
+			}
+
+			msg, err := info.asm.GenericMessage(ctx)
+			if err != nil {
+				logger.Warn().Msgf("could not convert to generic message %s", err.Error())
+				continue
+			}
+
+			err = IsProvisioningSource(ctx, info, sc)
+			if err != nil {
+				logger.Warn().Msgf("Provisioning app is not enabled in source %s", info.asm.SourceID)
```

Comment:
Can we include the Error here? I think there is valuable info in it, I believe there is the the info about what application_types there was found, so we can figure out what we get send and maybe limit the miss send checks in the future.

```suggestion
				logger.Warn().Err(err).Msgf("Provisioning app is not enabled in source %s", info.asm.SourceID)
```